tests/unit_tests/test_coverage.py

```diff
@@ -20,10 +23,62 @@ def fixture_metagenome(variables, tmp_path):
     return fpath.as_posix()
 
 
-# TODO: Create fixtures for each set of data.
-# Then group fixtures into a group and indirectly parametrize group
-# i.e. alignments.sam, alignments.bam, alignments.bed, fwd_reads.fq, rev_reads.fq
-# The external tools could be run or we could monkeypatch these.
+@pytest.fixture(name="sam_alignment")
+def fixture_alignment_sam(variables, tmp_path):
+    coverage_test_data = variables["coverage"]
+    outlines = coverage_test_data["sam"]
+    fpath = tmp_path / "records.sam"
+    with open(fpath, "w") as fh:
+        fh.write(outlines)
+    return fpath.as_posix()
+
+
+@pytest.fixture(name="bam_alignment")
+def fixture_alignment_bam(variables, sam_alignment, tmp_path):
+    bam_fpath = tmp_path / "records.bam"
+    samtools.sort(sam=sam_alignment, bam=bam_fpath)
+    return bam_fpath.as_posix()
```

Comment:
Setting up a fixture with an external dependency seems a bit precarious. We could leave this for now (assuming this works) but I can imagine a case where somewhere the external `samtools.sort` gets broken and now this fixture is broken for the rest of the coverage test suite. It seems like we should separately test `samtools.sort` in `test_samtools.py` and monkeypatch the calls to each of the external tools.

Reply:
I agree that in case `samtool.sort` gets brocken it will disrupt the entire fixture. However, this is the only way I could find to write a `bam` file in the most memory efficient manner.